lib/fhc.js

```diff
@@ -128,9 +128,15 @@ fhc.applyCommandFunction = function(processArgv, cb) {
 
   // Only new-style commands can be validated
   if (cmdFn.demand) {
+    if (cmdFn.deprecation) {
+      console.warn(colors.yellow(cmdFn.deprecation +"\n"));
+    }
```

Comment:
maybe we can move this to be just above L117 so that the deprecation notice will always printed? Also it means we don't need to print it again at the end?

Reply:
Yes. we can move this line for out of the if. However, I would like to keep the same code when it has an error too because some commands have a big body as it one and the user cannot see the warn message.

Reply:
Hi @wei-lee,

Few commands has a big body and the WARN message can be not faced as in this case. Because of it IHMO is better print on top and in the validation erros too always. Following images to ilustrate this. 

<img width="1350" alt="screen shot 2018-09-27 at 17 14 58" src="https://user-images.githubusercontent.com/7708031/46159721-e47c1000-c278-11e8-8c92-19f3aded1f64.png">


<img width="1070" alt="screen shot 2018-09-27 at 17 12 41" src="https://user-images.githubusercontent.com/7708031/46159689-d0381300-c278-11e8-8061-df39ffe1e280.png">

The first output is inside of the demand to ensure that will be called just when the command is executed. (In order to avoid some wird scenario, hwoever, it did not matter we can put out of the if if you prefer).